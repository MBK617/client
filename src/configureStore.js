/**
 * @file Configure redux store
 */

import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';

import createReducer from './reducers';

/**
 * Configures and returns initial redux store
 *
 * @param {object} initialState The initial state of the application
 * 
 * @return {object} A redux store
 * 
 * @memberof module:Main
 *
 */
function configureStore(initialState = {}) {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    createReducer(),
    initialState,
    applyMiddleware(sagaMiddleware)
  );

  store.runSaga = sagaMiddleware.run;
  store.injectedReducers = {};
  store.injectedSagas = {};

  return store;
}

export default configureStore;