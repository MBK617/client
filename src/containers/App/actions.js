export const setToken = (token) => ({ type: 'SET_TOKEN', token });
export const setTokenSuccess = (token) => ({ type: 'SET_TOKEN_SUCCESS', token });
export const getToken = () => ({ type: 'GET_TOKEN' });
export const getTokenSuccess = (token) => ({ type: 'GET_TOKEN_SUCCESS', token });

export const pushHistory = (path) => ({ type: 'PUSH_HISTORY', path });
export const popHistory = () => ({ type: 'POP_HISTORY' });