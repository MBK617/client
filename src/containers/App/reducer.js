/*
 * AppReducer
 */

import { fromJS } from 'immutable';

export const initialState = fromJS({
  token: null,
}).set('history', ['home']);

function appReducer(state = initialState, action) {
  switch (action.type) {
    case 'GET_TOKEN_SUCCESS':
      return state.set('token', action.token);
    case 'SET_TOKEN_SUCCESS':
      return state.set('token', action.token);
    case 'PUSH_HISTORY':
      return state.set('history', [action.path, ...state.get('history')]);
    case 'POP_HISTORY':
      return state.set('history', state.get('history').slice(1));
    default:
      return state;
  }
}

export default appReducer;