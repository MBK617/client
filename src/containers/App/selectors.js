import { createSelector } from "reselect";

export const selectApp = createSelector(
  state => state.app,
  substate => substate.toJS()
);