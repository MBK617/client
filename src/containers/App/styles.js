import { StyleSheet } from 'react-native'
import Constants from 'expo-constants'

const styles = StyleSheet.create({
  root: {
    marginTop: Constants.statusBarHeight,
    alignSelf: 'center',
    flexShrink: 0,
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'center',
    width: '100%',
  },
  header: {
    display: 'flex',
    justifyContent: 'space-between', 
    backgroundColor: '#fff'
  },
  logo: {
    width: 120,
    height: 25,
  }
});

export default styles;