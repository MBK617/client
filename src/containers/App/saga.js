import { all, takeLatest, put } from "redux-saga/effects";
import * as storage from 'utils/storage';
import { getTokenSuccess, setTokenSuccess } from "./actions";

export function* setTokenInStorage(action) {
  try {
    if(action.token) {
      yield storage.setItem('token', action.token);
    } else {
      yield storage.deleteItem('token');
    }
    yield put(setTokenSuccess(action.token));
  } catch (err) {
    console.error(err);
  }
}

export function* getTokenFromStorage() {
  try {
    const value = yield storage.getItem('token');
    if(value !== null) {
      yield put(getTokenSuccess(value));
    }
  } catch(err) {
    console.error(err);
  }
}

export default function* rootSaga() {
  yield all([
    takeLatest('SET_TOKEN', setTokenInStorage),
    takeLatest('GET_TOKEN', getTokenFromStorage),
  ]);
}