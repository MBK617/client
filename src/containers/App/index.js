import React from 'react';
import { View } from 'react-native';

import Counter from 'containers/Counter';

import styles from './styles';

function App() {
  return (
    <View style={styles.container}>
      <Counter/>
    </View>
  );
}

export default App;