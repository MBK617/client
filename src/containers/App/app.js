import React, { useEffect } from 'react';
import { View, Image } from 'react-native';
import { compose } from 'redux';

import { 
  Appbar, 
  BottomNavigation 
} from 'react-native-paper';

import injectSaga from 'utils/redux/injectSaga';
import injectReducer from 'utils/redux/injectReducer';

import useHistory from 'hooks/useHistory';
import useLoginStatus from 'hooks/useLoginStatus';

import styles from './styles';
import reducer from './reducer';
import saga from './saga';

import HomePage from '../HomePage';
import EventsPage from '../EventsPage';
import InboxPage from '../InboxPage';
import MorePage from '../MorePage';

const pages = [
  { key: 'home', icon: 'home', authenticated: false },
  { key: 'events', icon: 'calendar', authenticated: true },
  { key: 'inbox', icon: 'forum', authenticated: true },
  { key: 'more', icon: 'dots-horizontal', authenticated: false },
]

const App = () => {
  const isLoggedIn = useLoginStatus();
  const { history, path, goTo, goBack } = useHistory();

  const [index, setIndex] = React.useState(0);
  const [routes, setRoutes] = React.useState(pages.filter(({authenticated}) => isLoggedIn || !authenticated).map(({ authenticated, ...page }) => page));

  useEffect(() => {
    const newRoutes = pages.filter(({authenticated}) => isLoggedIn || !authenticated).map(({ authenticated, ...page }) => page);
    setIndex(newRoutes.findIndex(route => route.key === routes[index].key))
    setRoutes(newRoutes);
  }, [isLoggedIn])

  useEffect(() => {
    const newPath = routes[index].key;
    goTo(newPath);
  }, [index])

  useEffect(() => {
    const newIndex = routes.findIndex(page => page.key === path?.split('/')[0]);
    setIndex(newIndex);
  }, [path])

  const renderScene =  () => {
    switch (path?.split('/')[0]) {
      case 'home':
        return <HomePage />;
      case 'events':
        return <EventsPage />;
      case 'inbox':
        return <InboxPage />;
      case 'more':
        return <MorePage />;
    }
  }
  
  return (
    <View style={styles.root}>
      <Appbar.Header 
        statusBarHeight={0}
        style={styles.header}
      > 
        <Appbar.BackAction onPress={goBack} style={{opacity: history.length > 1 ? 1 : 0, transition: '.2s'}}/>
        <Image source={require('assets/mbklogo.jpg')} style={styles.logo}/>
        <Appbar.Action icon="bell" />
      </Appbar.Header>
      <BottomNavigation
        barStyle={{ backgroundColor: '#222' }}
        navigationState={{ index, routes }}
        onIndexChange={setIndex}
        renderScene={renderScene}
        shifting={false} 
        labeled={false} 
      />
    </View>
  );
}

const withReducer = injectReducer({ key: 'app', reducer });
const withSaga = injectSaga({ key: 'app', saga });

export default compose(withReducer, withSaga)(App);