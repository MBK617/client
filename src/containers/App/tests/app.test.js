import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render, cleanup, act } from '@testing-library/react-native';
import { fromJS } from 'immutable';
import createSagaMiddleware from 'redux-saga';
import useHistory from 'hooks/useHistory';
import useLoginStatus from 'hooks/useLoginStatus';

import App from '..';

jest.mock('@expo/vector-icons');

jest.mock('hooks/useHistory');

jest.mock('hooks/useLoginStatus');

const initialState = {
  app: fromJS({
    token: null,
  }).set('history', ['home'])
};

describe('<App />', () => {
  const sagaMiddleware = createSagaMiddleware();
  let store, rendered;

  const mockStore = (initialState) => ({ 
    ...configureStore([sagaMiddleware])(initialState), 
    injectedReducers: {}, 
    injectedSagas: {}, 
    runSaga: sagaMiddleware.run 
  });

  let goTo, goBack;

  describe('when logged in to home page', () => {
    beforeEach(() => {
      store = mockStore(initialState);
      goTo = jest.fn();
      goBack = jest.fn();
      useHistory.mockImplementation(() => ({
        history: ['home'],
        path: 'home',
        goTo,
        goBack
      }));
      useLoginStatus.mockImplementation(() => true);
      rendered = render(
        <Provider store={store}>
          <App />
        </Provider>
      );
    })
  
    afterEach(cleanup);
  
    it('should render without crashing', () => {
      const tree = rendered.toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('should display back button', () => {
      expect(rendered.getByLabelText(/Back/i)).toBeTruthy();
    });
  
    it('should disable back button without prior navigation', () => {
      const backButton = rendered.getByLabelText(/Back/i);
      expect(backButton.props.accessibilityState.disabled).toBeTruthy();
    });
    
    it('should call goBack on back click, provided prior navigation', () => {
      const inboxButton = rendered.getByLabelText(/Inbox/i);
      act(inboxButton.props.onClick);
      expect(goTo).toHaveBeenCalledWith('inbox')
      const backButton = rendered.getByLabelText(/Back/i);
      act(backButton.props.onClick);
      expect(goBack).toHaveBeenCalled();
    });
  
    it('should render bottom navigation when logged out', () => {
      useLoginStatus.mockImplementation(() =>  false);
      rendered = render(
        <Provider store={store}>
          <App />
        </Provider>
      );
      
      const buttons = rendered.queryAllByRole('button').slice(2);
      expect(buttons.map(b => b.props.accessibilityLabel)).toEqual(['Home', 'More']);
    })
  
    it('should render bottom navigation when logged in', () => {
      const buttons = rendered.queryAllByRole('button').slice(2);
      expect(buttons.map(b => b.props.accessibilityLabel)).toEqual(['Home', 'Events', 'Inbox', 'More']);
    })
  
    it('should navigate on bottom navigation clicks', () => {
      const buttons = rendered.queryAllByRole('button').slice(2);
      buttons.forEach(button => {
        act(button.props.onClick);
        expect(goTo).toHaveBeenCalledWith(button.props.accessibilityLabel.toLowerCase());
      })
    });
  });

  describe('events', () => {
    beforeEach(() => {
      store = mockStore(initialState);
      useLoginStatus.mockImplementation(() => true);
      useHistory.mockImplementation(() => ({
        history: ['events', 'home'],
        path: 'events',
        goTo,
        goBack
      }));
      rendered = render(
        <Provider store={store}>
          <App />
        </Provider>
      );
    });
  
    afterEach(cleanup);
  
    it('should render without crashing', () => {
      const tree = rendered.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('inbox', () => {
    beforeEach(() => {
      store = mockStore(initialState);
      useLoginStatus.mockImplementation(() => true);
      useHistory.mockImplementation(() => ({
        history: ['inbox', 'home'],
        path: 'inbox',
        goTo,
        goBack
      }));
      rendered = render(
        <Provider store={store}>
          <App />
        </Provider>
      );
    });
  
    afterEach(cleanup);
  
    it('should render without crashing', () => {
      const tree = rendered.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('more', () => {
    beforeEach(() => {
      store = mockStore(initialState);
      useLoginStatus.mockImplementation(() => true);
      useHistory.mockImplementation(() => ({
        history: ['more', 'home'],
        path: 'more',
        goTo,
        goBack
      }));
      rendered = render(
        <Provider store={store}>
          <App />
        </Provider>
      );
    });
  
    afterEach(cleanup);
  
    it('should render without crashing', () => {
      const tree = rendered.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});