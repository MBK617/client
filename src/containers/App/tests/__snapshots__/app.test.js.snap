// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexShrink": 0,
      "justifyContent": "center",
      "marginTop": 44,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "display": "flex",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        accessibilityLabel="Back"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              Object {
                "opacity": 0.32,
              },
              Object {
                "opacity": 0,
                "transition": ".2s",
              },
            ],
          ]
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                      "width": 24,
                    },
                  ]
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "resizeMode": "contain",
                      },
                      Object {
                        "height": 21,
                        "tintColor": "#000000",
                        "width": 21,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "width": 120,
          }
        }
      />
      <View
        accessibilityLabel="Notifications"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#f6f6f6",
          },
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        collapsable={false}
        importantForAccessibility="auto"
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "top": 0,
            }
          }
        >
          <View>
            <Text>
              Home
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "#222",
          "bottom": 0,
          "elevation": 4,
          "left": 0,
          "position": null,
          "right": 0,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#222",
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "width": "100%",
              },
              Object {
                "marginBottom": 0,
                "maxWidth": 384,
              },
            ]
          }
        >
          <View
            accessibilityLabel="Home"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "selected": true,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "paddingVertical": 6,
              }
            }
          >
            <View
              pointerEvents="none"
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 24,
                    "marginHorizontal": 12,
                    "marginTop": 2,
                    "transform": Array [
                      Object {
                        "translateY": 7,
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "left": 0,
                        "position": "absolute",
                        "top": -2,
                      },
                      Object {
                        "right": -2,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "alignSelf": "flex-end",
                        "backgroundColor": "#f50057",
                        "borderRadius": 8,
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontSize": 8,
                        "fontWeight": "400",
                        "height": 16,
                        "lineHeight": 16,
                        "minWidth": 16,
                        "opacity": 0,
                        "overflow": "hidden",
                        "paddingHorizontal": 4,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": 16,
                    "paddingBottom": 2,
                  }
                }
              />
            </View>
          </View>
          <View
            accessibilityLabel="Events"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "selected": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "paddingVertical": 6,
              }
            }
          >
            <View
              pointerEvents="none"
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 24,
                    "marginHorizontal": 12,
                    "marginTop": 2,
                    "transform": Array [
                      Object {
                        "translateY": 7,
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "left": 0,
                        "position": "absolute",
                        "top": -2,
                      },
                      Object {
                        "right": -2,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "alignSelf": "flex-end",
                        "backgroundColor": "#f50057",
                        "borderRadius": 8,
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontSize": 8,
                        "fontWeight": "400",
                        "height": 16,
                        "lineHeight": 16,
                        "minWidth": 16,
                        "opacity": 0,
                        "overflow": "hidden",
                        "paddingHorizontal": 4,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": 16,
                    "paddingBottom": 2,
                  }
                }
              />
            </View>
          </View>
          <View
            accessibilityLabel="Inbox"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "selected": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "paddingVertical": 6,
              }
            }
          >
            <View
              pointerEvents="none"
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 24,
                    "marginHorizontal": 12,
                    "marginTop": 2,
                    "transform": Array [
                      Object {
                        "translateY": 7,
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "left": 0,
                        "position": "absolute",
                        "top": -2,
                      },
                      Object {
                        "right": -2,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "alignSelf": "flex-end",
                        "backgroundColor": "#f50057",
                        "borderRadius": 8,
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontSize": 8,
                        "fontWeight": "400",
                        "height": 16,
                        "lineHeight": 16,
                        "minWidth": 16,
                        "opacity": 0,
                        "overflow": "hidden",
                        "paddingHorizontal": 4,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": 16,
                    "paddingBottom": 2,
                  }
                }
              />
            </View>
          </View>
          <View
            accessibilityLabel="More"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "selected": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "paddingVertical": 6,
              }
            }
          >
            <View
              pointerEvents="none"
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 24,
                    "marginHorizontal": 12,
                    "marginTop": 2,
                    "transform": Array [
                      Object {
                        "translateY": 7,
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "left": 0,
                        "position": "absolute",
                        "top": -2,
                      },
                      Object {
                        "right": -2,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "alignSelf": "flex-end",
                        "backgroundColor": "#f50057",
                        "borderRadius": 8,
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontSize": 8,
                        "fontWeight": "400",
                        "height": 16,
                        "lineHeight": 16,
                        "minWidth": 16,
                        "opacity": 0,
                        "overflow": "hidden",
                        "paddingHorizontal": 4,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": 16,
                    "paddingBottom": 2,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
